<launch>
  <arg name="DEBUG" default="false"/>
  <arg name="uav_name" default="$(optenv UAV_NAME uav1)"/>
  <arg name="gui" default="true"/>
  <arg name="method" default="3"/>
  <arg name="SamplePointSize" default="16"/>
  <arg name="ScanRadius" default="8"/>
  <arg name="StepSize" default="10"/>
  <arg name="FrameSize" default="1000"/>
  <arg name="publish" default="true"/>
  <arg name="useOdom" default="false"/>
  <arg name="cameraRotated" default="false"/>
  <arg name="FromVideo" default="false"/>
  <arg name="VideoNumber" default="1"/>
  <arg name="cellSize" default="64"/>
  <arg name="cellOverlay" default="16"/>
  <arg name="surroundRadius" default="3"/>
  <arg name="camNum" default="0"/>

  <group ns="$(arg uav_name)">

    <node name="drone_detector" pkg="mask_maker" type="mask_maker_node" output="screen"> 
      <param name="DEBUG"		type="bool" value="$(arg DEBUG)"/> 
      <param name="gui"             type="bool" value="$(arg gui)"/>
      <param name="FromVideo"         type="bool" value="$(arg FromVideo)"/>
      <param name="VideoNumber"     type="int" value="$(arg VideoNumber)"/>
      <param name="FrameSize"       type="int" value="$(arg FrameSize)"/>
      <param name="SamplePointSize" type="int" value="$(arg SamplePointSize)"/>
      <param name="ScanRadius"    type="int" value="$(arg ScanRadius)"/>
      <param name="StepSize"	type="int" value="$(arg StepSize)"/>
      <param name="cellSize"	type="int" value="$(arg cellSize)"/>
      <param name="cellOverlay"	type="int" value="$(arg cellOverlay)"/>
      <param name="surroundRadius"	type="int" value="$(arg surroundRadius)"/>
      <param name="method"	type="int" value="$(arg method)"/>
      <param name="publish"         type="bool" value="$(arg publish)"/>
      <param name="ScaleFactor"	type="int" value="1"/>
      <param name="RangerDataPath"    type="string" value="/$(arg uav_name)/terarangerone/range"/>
      <param name="CameraImageCompressed"    type="bool" value="false"/>
      <param name="cameraRotated"    type="bool" value="$(arg cameraRotated)"/>
      <param name="useOdom"           type="bool" value="$(arg useOdom)"/>
      <param name="camNum"           type="int" value="$(arg camNum)"/>



    </node>
  </group>
</launch>
